File: MapShapes.Data/Entities/OverlayShape.cs
namespace MapShapes.Data.Entities
{
    public class OverlayShape : KeyInt32
    {
        protected OverlayShape()
        {
        }

        public OverlayShape(string title, string properties, ShapeType type)
        {
            this.Title = title;
            this.Type = type;
            this.Properties = properties;
        }

        public string Properties { get; private set; }
        public string Title { get; private set; }

        public ShapeType Type { get; private set; }

        public void Edit(string title, string properties, ShapeType type)
        {
            this.Title = title;
            this.Properties = properties;
            this.Type = type;
        }
    }
}

File: MapShapes.Core.Api/angular-client/src/app/models/overlay-shape.ts
export interface OverlayShape {
  id: number;
  title: string;
  properties: string;
  type: string;
}

export class OverlayShape implements OverlayShape {
  constructor(type: string, title: string, properties: string, id: number = 0) {
    this.title = title;
    this.properties = properties;
    this.id = id;
    this.type = type;
  }
}